### DISCUSSION TITLE
Using PyCharm debug mode and breakpoints with pytorch-lightning?

### DISCUSSION THREAD

**Discussion by jpilaul** [Lightning-AI/pytorch-lightning#8810] [DDP / multi-GPU / multi-node]
How can we use PyCharm debugging mode on pytorch-lightning. I am using the latest pytorch-lightning version. At the moment, when I use debug mode, I am unable to view `Variables`. I think that it's related to a distributed training bug in PyCharm/pytorch debugging mode (https://youtrack.jetbrains.com/issue/PY-39489). It first says "Collecting data" and then I see the following:
![Screen Shot 2021-08-07 at 8 08 03 AM](https://user-images.githubusercontent.com/614861/128703784-62f5cb15-83c1-4dcd-b234-567c51f50a5a.png)
I have tried setting `gpus=1` but it still hangs at "Collecting data".

**Comment by erikhuck** [comment]
Big +1 for this. It's very tedious to have to use print statements within my lightning code rather than simply using the PyCharm debugger. Why is this happening???

**Comment by walsharry** [comment]
I had the same problem. The solution was to set num_workers to 0 while creating the DataLoader object, then it works like normal.

**Comment by LindaSt** [comment]
I'm having the same issue
